Remove unnecessary IOKit sandbox rule

pvollan opened the pull request:
#### 631491c3f80cbd49d14196b4b78c06c1255ea829
<pre>
Remove unnecessary IOKit sandbox rule
<a href="https://bugs.webkit.org/show_bug.cgi?id=253852">https://bugs.webkit.org/show_bug.cgi?id=253852</a>
rdar://problem/106668157

Reviewed by Sihui Liu.

Remove unnecessary IOKit sandbox rule in the WebContent process sandbox when
ENABLE(WEBCONTENT_GPU_SANDBOX_EXTENSIONS_BLOCKING) = 1.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:

Canonical link: <a href="https://commits.webkit.org/261717@main">https://commits.webkit.org/261717@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/de2a5abd6278b7ad69ddbfece152160427768bf7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/112272 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21409 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/903 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4064 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120901 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22752 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/12718 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/4835 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/118035 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/16908 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100088 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/105330 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/98865 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/629 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/45919 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13799 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/672 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/94260 "run-api-tests-without-change (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/14486 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/10159 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/19842 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52676 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/16283 "Built successfully") | | | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/4470 "Failed to push commit to Webkit repository") | | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -879,12 +879,12 @@
         (iokit-connection "IOGPU")))
-#endif
 
 (deny iokit-open-user-client
```
Should this be allow instead of deny after this change? (according to the rules above)

pvollan replied:
That is a good point. In this case, I think we do want to keep the deny, since we are only looking to remove this rule for the case when ENABLE(WEBCONTENT_GPU_SANDBOX_EXTENSIONS_BLOCKING) == 1. There should be no behavior change in either case, since this is denied by default.

Thanks for reviewing!

szewai reviewed (approved):
r=me

pvollan committed:
Remove unnecessary IOKit sandbox rule
https://bugs.webkit.org/show_bug.cgi?id=253852
rdar://problem/106668157

Reviewed by Sihui Liu.

Remove unnecessary IOKit sandbox rule in the WebContent process sandbox when
ENABLE(WEBCONTENT_GPU_SANDBOX_EXTENSIONS_BLOCKING) = 1.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:

Canonical link: https://commits.webkit.org/261717@main
